### PR TITLE
refactor: change some internal fields to ten_value_t and implement loop_all_fields functions

### LINKED ISSUE #105: Use ten_value_t uniformly to represent each field in the message
Use `ten_value_t` uniformly to represent each field in the message, including the built-in fields within `_ten`. This allows all fields to be looped through using a unified `for loop` in the protocol, and the unified serialization and deserialization methods of `ten_value_t` can be used for processing within the protocol plugin.

### LINKED ISSUE #155: Change the `to_json`/`from_json` of `msg` to use the `loop_all_fields` mechanism

### PR DESCRIPTION
**Pull request by sunxilin**
refactor: change some internal fields to ten_value_t & implement loop fields functions

### CODE REVIEW

**Comment by sunxilin** [comment]
This pull request includes several changes primarily focused on refactoring and improving the consistency of the codebase. The changes include renaming functions for clarity, adding new API functions, and updating field names and processing methods in various files.

### Function Renaming and Refactoring:

* [`core/include/ten_runtime/binding/cpp/internal/msg/audio_frame.h`](diffhunk://#diff-faefaa3a5dc1e33ec4995e0f086ee70150f986cbb8e02c9a9fa7ca0052ee2c7fL137-R137): Renamed `ten_audio_frame_alloc_data` to `ten_audio_frame_alloc_buf` for consistency.
* [`core/include/ten_runtime/msg/audio_frame/audio_frame.h`](diffhunk://#diff-d371445c7bdd7f12fc8a047e4645fcae0eac0638226795791372f583f7389fefL75-R75): Updated the function name `ten_audio_frame_alloc_data` to `ten_audio_frame_alloc_buf`.
* [`core/include_internal/ten_runtime/msg/audio_frame/audio_frame.h`](diffhunk://#diff-575e747aecba345ddd11cbf2937afee4408b99abcfebc3c8d80632ce2979bac7L41-R41): Changed the field name from `data` to `buf` in the `ten_audio_frame_t` struct.

### API Additions:

* [`core/include/ten_runtime/msg/cmd_result/cmd_result.h`](diffhunk://#diff-efb30a8372de959d2f6d774168dbaa91bc1bcac3640aa6c513c11a70ad7cac86R26-R32): Added new API functions `ten_raw_cmd_result_is_final` and `ten_raw_cmd_result_set_final`.
* [`core/include/ten_utils/value/value_json.h`](diffhunk://#diff-f46fc5fac1ad29d74ec0d45b6424b3aef8f02592147f726c0249b42cf2bae78dR20-R22): Introduced `ten_value_init_from_json` API function.
* [`core/include_internal/ten_runtime/common/loc.h`](diffhunk://#diff-12a44e15b08638953ad537331e883cb4f0aa6e90e0436e54756e5c7e8144b64eR82-R91): Added `ten_loc_to_value` and `ten_loc_create_from_value` API functions.

### Field and Method Updates:

* [`core/include_internal/ten_runtime/msg/audio_frame/field/field.h`](diffhunk://#diff-08d8e2c9e7b8e4cea837acf7e6437e23e1ced7a945480c8e5a2dc3d1eb1a9292L18-R18): Renamed `TEN_AUDIO_FRAME_FIELD_DATA` to `TEN_AUDIO_FRAME_FIELD_BUF`.
* [`core/include_internal/ten_runtime/msg/audio_frame/field/field_info.h`](diffhunk://#diff-7da9ef1925c9112e5de699e45ecbc93563f9433d3ec26c8ffa674ab97de2e947R36): Updated field processing methods to include `process_field` for various fields. [[1]](diffhunk://#diff-7da9ef1925c9112e5de699e45ecbc93563f9433d3ec26c8ffa674ab97de2e947R36) [[2]](diffhunk://#diff-7da9ef1925c9112e5de699e45ecbc93563f9433d3ec26c8ffa674ab97de2e947R45) [[3]](diffhunk://#diff-7da9ef1925c9112e5de699e45ecbc93563f9433d3ec26c8ffa674ab97de2e947R54) [[4]](diffhunk://#diff-7da9ef1925c9112e5de699e45ecbc93563f9433d3ec26c8ffa674ab97de2e947R65) [[5]](diffhunk://#diff-7da9ef1925c9112e5de699e45ecbc93563f9433d3ec26c8ffa674ab97de2e947R77) [[6]](diffhunk://#diff-7da9ef1925c9112e5de699e45ecbc93563f9433d3ec26c8ffa674ab97de2e947R88) [[7]](diffhunk://#diff-7da9ef1925c9112e5de699e45ecbc93563f9433d3ec26c8ffa674ab97de2e947R97-R111) [[8]](diffhunk://#diff-7da9ef1925c9112e5de699e45ecbc93563f9433d3ec26c8ffa674ab97de2e947R120)

### Miscellaneous Changes:

* [`.vscode/launch.json`](diffhunk://#diff-bd5430ee7c51dc892a67b3f2829d1f5b6d223f0fd48b82322cfd45baf9f5e945L140-R140): Modified the `--gtest_filter` argument to exclude `ExtensionTest.*AttemptToSuspend*`.
* [`core/include_internal/ten_runtime/common/constant_str.h`](diffhunk://#diff-25a6abdce4ed1762b2aea392a31dddbfe3a5432e338f6ff56150a5e8d7ae853bR33): Added new string constants `TEN_STR_IS_FINAL` and `TEN_STR_BUF`. [[1]](diffhunk://#diff-25a6abdce4ed1762b2aea392a31dddbfe3a5432e338f6ff56150a5e8d7ae853bR33) [[2]](diffhunk://#diff-25a6abdce4ed1762b2aea392a31dddbfe3a5432e338f6ff56150a5e8d7ae853bR116-R118)

These changes aim to improve code readability, maintainability, and consistency across the codebase.

File: core/src/ten_runtime/msg/cmd_base/cmd/start_graph/field/extension_info.c [hunk 1]
```diff
@@ -285,3 +285,13 @@ bool ten_cmd_start_graph_copy_extensions_info(
...
 }
+
+bool ten_cmd_start_graph_process_extensions_info(
+    ten_msg_t *self, ten_raw_msg_process_one_field_func_t cb, void *user_data,
+    ten_error_t *err) {
+  TEN_ASSERT(self && ten_raw_msg_check_integrity(self), "Should not happen.");
+
+  ten_cmd_start_graph_t *cmd = (ten_cmd_start_graph_t *)self;
+
+  // TODO(xilin): to be implemented.
```
Comment (sunxilin) [inline core/src/ten_runtime/msg/cmd_base/cmd/start_graph/field/extension_info.c]: Working in progress. Once implementing this method, we can change the to_json / from_json  of msg to use loop_all_fields mechanism.

File: core/include_internal/ten_runtime/msg/cmd_base/cmd/start_graph/cmd.h [hunk 2]
```diff
@@ -13,7 +13,7 @@
 typedef struct ten_cmd_start_graph_t {
   ten_cmd_t cmd_hdr;
 
-  bool long_running_mode;
+  ten_value_t long_running_mode;  // string
```
Comment (halajohn) [inline core/include_internal/ten_runtime/msg/cmd_base/cmd/start_graph/cmd.h]: 这个应该不是 string? 是 bool?

Comment (sunxilin) [reply core/include_internal/ten_runtime/msg/cmd_base/cmd/start_graph/cmd.h]: 没错 笔误

**Review by halajohn — APPROVED** [review]

**Pull request merged**